### DISCUSSION TITLE
Podman-compose tries to pull local images from docker registry

### DISCUSSION THREAD

**Discussion by rishacha** [containers/podman#23638] [Q&A]
My compose file - 

```yml
services:
  app:
      # image: python-pdm-dev
      build:
        context: .
        dockerfile: Dockerfile
      platform: linux/amd64
      stdin_open: true # docker run -i
      tty: true        # docker run -t
      env_file:
        - .env
      volumes:
        - ./:/app
```

Logs -

```
Successfully tagged localhost/test_app:latest
8ebe3d5b2fe6544d687452b256397540cffdb6163eb2c28a88f8bb2a943f78ac
9833ed7f2f8c8f7f28ab46039a80a73bf1c1dbfd1844c0ecd8c28528ff0c123a
cd5ef9a7bb45c539ae26d8e169770cff3cef630ef17ae7c379fd3df6acdce94b
Resolving "test_app" using unqualified-search registries (/etc/containers/registries.conf.d/999-podman-machine.conf)
Trying to pull docker.io/library/kafka-simulation_app:latest...
Error: initializing source docker://test_app:latest: reading manifest latest in docker.io/library/test_app: requested access to the resource is denied
Error: no container with name or ID "test_app_1" found: no such container
```

**Comment by rishacha** [comment]
@rhatdan - Need some help here

**Comment by rishacha** [comment]
Possible Solution - 

Solved by - @yuusou

1. SSH into the podman machine - `podman machine ssh`
2. Edit the following file `/etc/containers/registries.conf.d/999-podman-machine.conf` (`sudo nano /etc/containers/registries.conf.d/999-podman-machine.conf`)
3. It should look like this - 

```
# Issue #11489: make sure that we can inject a custom registries.conf
# file on the system level to force a single search registry.
# The remote client does not yet support prompting for short-name
# resolution, so we enforce a single search registry (i.e., [docker.io](http://docker.io/))
# as a workaround.
# unqualified-search-registries=["[docker.io](http://docker.io/)"]
# Change the above line to -

unqualified-search-registries=["localhost"]
```

4. Make the changes by adding only `localhost` in the array and removing `docker.io`
5. Save the file
6. Restart the machine

Voila !

**Reply by maysunfaisal** [reply]
I just installed Podman Desktop Version 1.19.1  on my new arm64 Macbook and come across this issue. The solution does not work for me unfortunately.

**Reply by rishacha** [reply]
@maysunfaisal - Was this fixed? Can you post some logs?

**Comment by nboaldin** [comment]
Same issue. I feel this should be added to podman compose docs or fixed.

**Comment by trane77** [comment]
I'm not sure if this fixed it or some other setting is required, as podman wasn't really a drop-in replacement as advertised, but maybe, it's just me not being expert enough.

BTW, try by uncommenting `prefix=...` in `/etc/containers/registries.conf` and setting it as `prefix = "docker.io"`, then do a compose up and see if it works.

If you're a noob like me, and using a Fedora/SELinux based distro, remember to add ':z' at the end of every local volume (where RW file access is needed) and remove volumes with `docker compose down --volumes` before restarting, otherwise the previous configuration is kept.

Oh, and follow [this guy's advice](https://github.com/laravel/sail/issues/83#issuecomment-3194582968) if you need to run Laravel Sail or similar setups where UID/GID must be shared with a container instance.

I'm not trying to be polemical or give OT information here, I just think the point of docker is making life easier... switching from environment setup to containerization environment setup, is just moving the problem to another context, not really solving it ☝️
